//#ifndef STEP_MACHINE_H
//#define STEP_MACHINE_H

//class stepMachine {

//public:

//    enum class Event { SagReached = 0, LatReached = 1, Start = 2, Stop = 3, Empty = 4 };
//    enum class State { Idle = 0, Walking = 1, Starting = 2, Stopping = 4, LastStep = 5 };


//    State update();
//    State getState();

//    bool setEvent();
//    bool reset();

//private:

//    Event _previous_event, _current_event;
//    State _previous_state, _current_state;

//    double _time, _new_event_time;



//}

//#endif // STEP_MACHINE_H
